lib/linguist/heuristics.rb

```diff
@@ -223,5 +223,30 @@ def call(data)
         Language["Text"]
       end
     end
+    disambiguate "PLSQL", "SQLPL", "PLpgSQL", "SQL" do |data|
+      match = nil 
+      #unique keywords for each language
+      plpgsqlkeywords = [/^\\i\b/i, /begin ?(work|transaction)?;/i , /AS \$\$/i, /RAISE EXCEPTION/i, /LANGUAGE plpgsql\b/i ]
+      plsqlkeywords = [ /pragma\b/i , /constructor\W+function\b/i]
+      #sqlplkeywords = [  ]
+
+      #common keywords that are not SQL
+      plkeywords = [/begin\b/i,/boolean\b/i, /package\b/i,/exception\b/i, /^end;\b/i ]
+
+      re = Regexp.union(plpgsqlkeywords)
+      if data.match(re)
+          match =  Language["PLpgSQL"] 
+      else
+         re = Regexp.union(plsqlkeywords)
+         if data.match(re)
```

Comment:
Related to [this](https://github.com/github/linguist/pull/2082/files#r24255160) it would be great to have this logic expressed in a single `if ... elsif ... end` block for clarity.